src/tools/perf/lib/libperf.c

```diff
@@ -894,7 +894,8 @@ static void ucp_perf_test_destroy_eps(ucx_perf_context_t* perf)
 {
     unsigned i, thread_count = perf->params.thread_count;
     unsigned num_in_prog     = 0;
-    ucs_status_ptr_t **reqs  = ucs_alloca(thread_count * sizeof(*reqs));
+    ucs_status_ptr_t **reqs  = ucs_alloca(2 * /* (ep + self_ep) */
```

Comment:
why need to allocate for self_ep?
there should be only one request for self_ep, so it could be a single local var

Reply:
We need a self_ep for each thread, don't we? As far as I understand m_perf is thread-local

Reply:
self_ep - yes, request - no